templates/app/aspnet-core/src/MyCompanyName.MyProjectName.Domain/Data/MyProjectNameDbMigrationService.cs

```diff
@@ -37,51 +37,45 @@ public async Task MigrateAsync()
         {
             Logger.LogInformation("Started database migrations...");
 
-            await MigrateHostDatabaseAsync();
+            await MigrateDatabaseAsync();
 
             var tenants = await _tenantRepository.GetListAsync(includeDetails: true);
 
-            var i = 0;
+            var migratedDatabases = new HashSet<string>();
             foreach (var tenant in tenants.Where(t => t.ConnectionStrings.Any()))
             {
-                using (_currentTenant.Change(tenant.Id))
+                var tenantConnectionStrings = tenant.ConnectionStrings.Select(x => x.Value).ToList();
+                
+                if (!migratedDatabases.Any() || !migratedDatabases.IsSupersetOf(tenantConnectionStrings))
                 {
-                    Logger.LogInformation($"Migrating {tenant.Name} database schema... ({++i} of {tenants.Count})");
-                    await MigrateTenantDatabasesAsync(tenant);
-                    Logger.LogInformation($"Successfully completed {tenant.Name} database migrations.");
+                    using (_currentTenant.Change(tenant.Id))
+                    {
+                        await MigrateDatabaseAsync(tenant);
+                    }
```

Comment:
If two tenants use the same connection string, the data of the second tenant will not be migrated.

Reply:
You need to combine the` tenant.Id` and connection string to determine whether the data has been migrated.